platform/generic_unix/wayland/context_egl_wayland.cpp

```diff
@@ -0,0 +1,204 @@
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <wayland-client-protocol.h>
+#include <wayland-client.h>
+#include <wayland-egl.h>
+#include <wayland-server.h>
+
+#include "context_egl_wayland.h"
+
+void ContextGL_EGL::release_current() {
+	eglMakeCurrent(egl_display, EGL_NO_SURFACE, EGL_NO_SURFACE, egl_context);
+}
+
+void ContextGL_EGL::make_current() {
+	eglMakeCurrent(egl_display, egl_surface, egl_surface, egl_context);
+}
+
+void ContextGL_EGL::swap_buffers() {
+	if (eglSwapBuffers(egl_display, egl_surface) != EGL_TRUE) {
+		cleanup();
+		initialize();
+		// tell rasterizer to reload textures and stuff?
+	}
+}
+
+void ContextGL_EGL::set_window_size(int width, int height) {
+	eglSwapBuffers(egl_display, NULL);
+	this->width = (EGLint)width;
+	this->height = (EGLint)height;
+}
+
+Error ContextGL_EGL::initialize() {
+	eglBindAPI(EGL_OPENGL_API);
+	EGLint configAttribList[] = {
+		EGL_RED_SIZE, 8,
+		EGL_GREEN_SIZE, 8,
+		EGL_BLUE_SIZE, 8,
+		// EGL_DEPTH_SIZE, 8,
+		// EGL_STENCIL_SIZE, 8,
+		// EGL_SAMPLE_BUFFERS, 0,
+		EGL_NONE
+	};
+	EGLint surfaceAttribList[] = {
+		EGL_NONE, EGL_NONE
+	};
+
+	EGLint numConfigs = 8;
+	EGLint majorVersion = 1;
+	EGLint minorVersion;
+	if (context_type == GLES_2_0) {
+		minorVersion = 0;
+	} else {
+		minorVersion = 5;
+	}
+	egl_display = EGL_NO_DISPLAY;
+	egl_context = EGL_NO_CONTEXT;
+	egl_surface = EGL_NO_SURFACE;
+	EGLConfig config = nullptr;
+
+	EGLint contextAttribs[3];
+	// if (context_type == GLES_2_0) {
+	// 	contextAttribs[0] = EGL_CONTEXT_MAJOR_VERSION;
+	// 	contextAttribs[1] = 3;
+	// 	contextAttribs[0] = EGL_CONTEXT_MINOR_VERSION;
+	// 	contextAttribs[1] = 3;
+	// 	contextAttribs[4] = EGL_NONE;
+	// } else {
+	// 	contextAttribs[0] = EGL_CONTEXT_MAJOR_VERSION;
+	// 	contextAttribs[1] = 3;
+	// 	contextAttribs[0] = EGL_CONTEXT_MINOR_VERSION;
+	// 	contextAttribs[1] = 3;
+	// 	contextAttribs[4] = EGL_NONE;
+	// }
```

Comment:
Could you also have a look at this section? I was just trying around what works.

Reply:
Sure thing.

Reply:
Took at look at this just now. We don't need to specify these things, and it's better to use the defaults here. However, while studying this code I found that we can use a more conservative `EGL_CONTEXT_CLIENT_VERSION`, and updated the EGL config accordingly.